sqlx-core/src/query_builder.rs

```diff
@@ -314,7 +314,13 @@ where
 
         let mut separated = self.separated(", ");
 
+        let mut empty_checker = false;
```

Comment:
It's actually not necessary to use an extra boolean here. You can just use `separated.push_separator` which will be `true` if any content was pushed.